inc/3rd-party/jetpack.php

```diff
@@ -1,56 +1,124 @@
+	) {
+		add_filter( 'rocket_first_install_options', '__rocket_add_jetpack_sitemap_option' );
+		function __rocket_add_jetpack_sitemap_option( $options ) {
+			$options['jetpack_xml_sitemap'] = 0;
+
+			return $options;
+		}
+
+		add_filter( 'rocket_inputs_sanitize', '__rocket_jetpack_sitemap_option_sanitize' );
+		function __rocket_jetpack_sitemap_option_sanitize( $inputs ) {
+			$inputs['jetpack_xml_sitemap'] = ! empty( $inputs['jetpack_xml_sitemap'] ) ? 1 : 0;
+
+			return $inputs;
+		}
+
+		add_filter( 'rocket_sitemap_preload_list', '__rocket_add_jetpack_sitemap' );
+		function __rocket_add_jetpack_sitemap( $sitemaps ) {
+			if ( get_rocket_option( 'jetpack_xml_sitemap', false ) ) {
+				$sitemaps['jetpack'] = jetpack_sitemap_uri();
+			}
+
+			return $sitemaps;
+		}
+
+		add_filter( 'rocket_sitemap_preload_options', '__rocket_sitemap_preload_jetpack_option' );
+		function __rocket_sitemap_preload_jetpack_option( $options ) {
+			$options[] = array(
+				'parent'       => 'sitemap_preload',
+				'type'         => 'checkbox',
+				'label'        => __( 'Jetpack XML Sitemaps', 'rocket' ),
+				'label_for'    => 'jetpack_xml_sitemap',
+				'label_screen' => sprintf( __( 'Preload the sitemap from the Jetpack plugin', 'rocket' ), 'Jetpack' ),
+				'default'      => 0,
+			);
+			$options[] = array(
+				'parent'        => 'sitemap_preload',
+				'type'          => 'helper_description',
+				'name'          => 'jetpack_xml_sitemap_desc',
+				'description'   => sprintf( __( 'We automatically detected the sitemap generated by the %s plugin. You can check the option to preload it.', 'rocket' ), 'Jetpack' ),
+			);
+
+			return $options;
+		}
+	} // End if().
+
+	/**
+	 * Support Jetpack's EU Cookie Law Widget.
+	 *
+	 * @see https://jetpack.com/support/extra-sidebar-widgets/eu-cookie-law-widget/
+	 *
+	 * @since 2.9.12
+	 */
+	if ( Jetpack::is_module_active( 'widgets' ) ) :
+
+		/**
+		 * Add the EU Cookie Law to the list of mandatory cookies before generating caching files.
+		 *
+		 * @param array $cookies List of mandatory cookies.
+		 */
+		function rocket_add_jetpack_cookie_law_mandatory_cookie( $cookies ) {
+			$cookies['jetpack-eu-cookie-law'] = 'eucookielaw';
+
+			return $cookies;
+		}
+		add_filter( 'rocket_cache_mandatory_cookies' , 'rocket_add_jetpack_cookie_law_mandatory_cookie' );
+
+		// Don't add the WP Rocket rewrite rules to avoid issues.
+		add_filter( 'rocket_htaccess_mod_rewrite', '__return_false' );
+
+		/**
+		 * Add cookies when we activate the plugin.
+		 */
+		function rocket_activate_jetpack_cookie_law() {
+			add_filter( 'rocket_htaccess_mod_rewrite'    , '__return_false' );
+			add_filter( 'rocket_cache_mandatory_cookies' , 'rocket_add_jetpack_cookie_law_mandatory_cookie' );
+
+			// Update the WP Rocket rules on the .htaccess file.
+			flush_rocket_htaccess();
+
+			// Regenerate the config file.
+			rocket_generate_config_file();
+		}
+		add_action( 'activate_jetpack/jetpack.php', 'rocket_activate_jetpack_cookie_law' );
```

Comment:
This isn't great as I'd like this code to run once, either when the widget is added, or when someone activates WP Rocket while our widget is already active, or when both are already active and they update to the next version of WP Rocket. I'll try to come up with something better.